setup.py

```diff
@@ -178,7 +178,7 @@ def recursive_data_files(directory, install_directory):
     package_data={
         "snapcraft": ["templates/*"],
     },
-    python_requires=">=3.10",
+    python_requires=">=3.12",
```

Comment:
This is breaking the trivy workflow: https://github.com/canonical/snapcraft/actions/runs/12199935616/job/34034945727?pr=5136#step:10:445

I'm wondering if I should revert this change.

Reply:
I'm curious where trivy gets py3.10 as its default. It seems like maybe `uv` is [selecting it here](https://github.com/canonical/starflow/blob/9c179fa633d896b32a9cbb76d70373995af627ec/.github/workflows/scan-python.yaml#L130) by default. Maybe that could be bumped?

Reply:
Looks like [ubuntu-latest](https://github.com/canonical/starflow/blob/main/.github/workflows/scan-python.yaml#L71) is still jammy, which is still Python 3.10
so I guess either we revert this or we parametrize the starflow workflow to accept a python version as input

Reply:
well, maybe sit on this [for a little while](https://github.com/actions/runner-images/issues/10636)

Reply:
Good idea, I'll wait